Sources/Hedera/Transaction.swift

```diff
@@ -693,3 +688,50 @@ private func protoTransactionBodyEqual(_ lhs: Proto_TransactionBody, _ rhs: Prot
 
     return true
 }
+
+protocol TransactionIdCollection {
+    associatedtype TransactionId
+    func append(_ item: TransactionId)
+    var count: Int { get }
+    func getTransactionId(at index: Int) -> TransactionId?
+
+    // You can add more methods as required.
+}
+
+class TransactionIdArray<T>: TransactionIdCollection {
+    private var array: [T] = []
+    private let lock = NSLock()
+
+    func append(_ item: T) {
+        lock.lock()
+        defer { lock.unlock() }
+        array.append(item)
+    }
+
+    var count: Int {
+        lock.lock()
+        defer { lock.unlock() }
+        return array.count
+    }
+
+    func getTransactionId(at index: Int) -> T? {
+        lock.lock()
+        defer { lock.unlock() }
+        guard index >= 0 && index < array.count else { return nil }
+        return array[index]
+    }
+
+    func performLocked(_ closure: () -> Void) {
+        lock.lock()
+        defer { lock.unlock() }
+        closure()
+    }
+
+    var isEmpty: Bool {
+        lock.lock()
+        defer { lock.unlock() }
+        return array.isEmpty
+    }
+
+    // Implement other protocol methods here.
+}
```

Comment:
It seems like this whole thing is unused, and even if it isn't, I think this is not the way
We have well designed locks already from NIOConcurrencyHelpers